src/lein_monolith/plugin.clj

```diff
@@ -4,76 +4,93 @@
 
 
-(defn- add-profile-paths
-  "Update a profile paths entry by adding the absolute paths from the given
-  project. Returns the updated profile."
-  [profile project k]
-  (update profile k into
-          (map (partial str (:root project) "/")
-               (get project k))))
+(defn- maybe-mark-leaky
+  "Add ^:leaky metadata to a profile if it is of the leaky type."
+  [profile types]
+  (cond-> profile
+    (:leaky types) (vary-meta assoc :leaky true)))
 
 
-(defn merged-profile
-  "Constructs a profile map containing merged (re)source and test paths."
-  [subprojects]
-  (reduce-kv
-    (fn [profile _ project]
-      (-> profile
-          (add-profile-paths project :resource-paths)
-          (add-profile-paths project :source-paths)
-          (add-profile-paths project :test-paths)))
-    {:dependencies (subproject-dependencies subprojects)
-     :resource-paths []
-     :source-paths []
-     :test-paths []}
-    subprojects))
+(defn- choose-inheritance-source
+  "Choose either the initialized monolith or its raw representation for use when
+  building an inherited profile."
+  [monolith types]
+  (if (:raw types)
+    (:monolith/raw (meta monolith))
+    monolith))
 
 
 (defn- select-inherited-properties
-  "Constructs a profile map containing the given inherited properties from the
-  monolith project map."
-  [monolith base-properties setting]
-  (cond
-    ;; Don't inherit anything
-    (not setting)
-    nil
-
-    ;; Inherit the base properties specified in the parent.
-    (true? setting)
-    (select-keys monolith base-properties)
-
-    ;; Provide additional properties to inherit, or replace if metadata is set.
-    (vector? setting)
-    (select-keys monolith
-                 (if (:replace (meta setting))
-                   setting
-                   (distinct (concat base-properties setting))))
-
-    :else
-    (throw (ex-info (str "Unknown value type for monolith inherit setting: "
-                         (pr-str setting))
-                    {:inherit setting}))))
+  "Constructs a profile map containing the inherited properties from a parent
+  project map."
+  [monolith base-properties subproject subproject-key]
+  (let [default (boolean (:monolith/inherit subproject))
+        setting (subproject-key subproject default)]
+    (cond
+      ;; Don't inherit anything
+      (not setting)
+      nil
+
+      ;; Inherit the base properties specified in the parent.
+      (true? setting)
+      (select-keys monolith base-properties)
+
+      ;; Provide additional properties to inherit, or replace if metadata is set.
+      (vector? setting)
+      (select-keys monolith
+                   (if (:replace (meta setting))
+                     setting
+                     (distinct (concat base-properties setting))))
+
+      :else
+      (throw (ex-info (str "Unknown value type for monolith inherit setting: "
+                           (pr-str setting))
+                      {:inherit setting})))))
```

Comment:
Would be good to include the `subproject-key` in the message/ex-data here, now that it's in scope.